### PR TITLE
WIP: New packages: ta-lib-0.4.0, python-ta-lib-0.4.17

### PR DESCRIPTION
**Pull request by jpmenil**

### CODE REVIEW

**Comment by Chocimier** [comment]
Why do we need library? Any application will use it?

**Comment by jpmenil** [comment]
@Chocimier i'm preparing a PR for python-ta-lib.

File: srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+   ta_abstract/libta_abstract.la \
+   ta_func/libta_func.la \
+-  ta_common/libta_common.la
++  ta_common/libta_common.la -lm
+ 
+ libta_lib_la_LDFLAGS = -version-info $(TALIB_LIBRARY_VERSION)
+ 
+diff --git a/src/Makefile.in b/src/Makefile.in
+index 489b44a..ef1f506 100644
+--- a/src/Makefile.in
```
Comment (Chocimier) [inline srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch]: Is it your patch? if so, please make it -Np0

File: srcpkgs/ta-lib/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'ta-lib'
+pkgname=ta-lib
+version=0.4.0
+revision=1
+wrksrc="${pkgname}"
```
Comment (Chocimier) [inline srcpkgs/ta-lib/template]: just ta-lib, here and later

File: srcpkgs/ta-lib/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+build_style=configure
+configure_args="--prefix=/usr"
+short_desc="Technical Analysis Library for analyzing financial markets trend"
+maintainer="Jean-Philippe Menil <jpmenil@gmail.com>"
+license="custom:TA-Lib BSD License"
```
Comment (Chocimier) [inline srcpkgs/ta-lib/template]: It's 3-clause

File: srcpkgs/ta-lib/template [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+homepage="http://ta-lib.org"
```
Comment (Chocimier) [inline srcpkgs/ta-lib/template]: https please

File: srcpkgs/ta-lib/template [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+distfiles="${SOURCEFORGE_SITE}/${pkgname}/${pkgname}-${version}-src.tar.gz"
+checksum=9ff41efcb1c011a4b4b6dfc91610b06e39b1d7973ed5d4dee55029a0ac4dc651
+patch_args="-Np1"
+
+archs="i686 x86_64 x86_64-musl"
```
Comment (Chocimier) [inline srcpkgs/ta-lib/template]: Setting `archs` means something specific to arch, not a case here.  Set `nocross` with reason instead. I will also see if cross-compilation can be fixed.

File: srcpkgs/ta-lib/template [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+do_build() {
```
Comment (Chocimier) [inline srcpkgs/ta-lib/template]: Default do_build and do_install looks to fit, please move vlicense to post_install if I'm right.

File: srcpkgs/ta-lib/files/LICENSE [hunk 7]
```diff
@@ -0,0 +1,31 @@
+TA-Lib Copyright (c) 1999-2007, Mario Fortier
```
Comment (Chocimier) [inline srcpkgs/ta-lib/files/LICENSE]: Don't put license file into repo, either download or generate during build, see moreutils template.

File: srcpkgs/ta-lib/template (hunk 3 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: thx, was really unsure about this one

File: srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch (hunk 1 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch]: not mine, took this one from gentoo

File: srcpkgs/ta-lib/template (hunk 6 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: seem that config.guess is out of date, i'm finally able to build on aarch64.
Dunno how i will manage that in the template, maybe a case for arm?

File: srcpkgs/ta-lib/template (hunk 5 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: seem that config.guess is out of date, i'm finally able to build on aarch64.
Dunno how i will manage that in the template, maybe a case for arm?

File: srcpkgs/ta-lib/template (hunk 4 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: ack

File: srcpkgs/ta-lib/template (hunk 2 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: ack

File: srcpkgs/ta-lib/files/LICENSE (hunk 7 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/files/LICENSE]: ack

File: srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch (hunk 1 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch]: i took it because of the -lm, but seem it's finally not needed.

**Comment by jpmenil** [comment]
> > preparing a PR for python-ta-lib.
> 
> Ok, best if they are in one PR, one commit per template.
> 
> Please split ta-lib-devel.

just to be sure to understand, you want me to add both python-ta-lib and ta-lib-devel in the same PR, right ?

File: srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch (hunk 1 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/patches/ta-lib-0.4.0-asneeded.patch]: builds break without the patch ...

File: srcpkgs/ta-lib/template (hunk 5 above)
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: i made a case for arm/aarch.
Is there any way to do a hostmakedepends regarding the arch ?

**Comment by jpmenil** [comment]
Build failed is expected since python-ta-lib cannot be build without ta-lib installed.
However, i'm stuck with the arm/aarch builds, even if it compile fine on my local machines.

**Comment by Chocimier** [comment]
* compiled part, so isn't noarch

**Comment by Chocimier** [comment]
Also please set nocross="python3-numpy is nocross" on module.

File: srcpkgs/ta-lib/template [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+homepage="https://ta-lib.org"
+distfiles="${SOURCEFORGE_SITE}/ta-lib/ta-lib-${version}-src.tar.gz"
+checksum=9ff41efcb1c011a4b4b6dfc91610b06e39b1d7973ed5d4dee55029a0ac4dc651
+disable_parallel_build=yes
+patch_args="-Np1"
+
+pre_build() {
+	case "$XBPS_TARGET_MACHINE" in
+	armv*|aarch*)
+		curl -o config.guess "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD"
```
[the remark this answers is not in the document]
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: ack

File: srcpkgs/ta-lib/template [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# Template file for 'ta-lib'
+pkgname=ta-lib
+version=0.4.0
+revision=1
+wrksrc="ta-lib"
+build_style=configure
+configure_args="--prefix=/usr"
+hostmakedepends="curl"
```
[the remark this answers is not in the document]
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: ack

File: srcpkgs/ta-lib/template [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# Template file for 'ta-lib'
+pkgname=ta-lib
+version=0.4.0
+revision=1
+wrksrc="ta-lib"
+build_style=configure
+configure_args="--prefix=/usr"
```
[the remark this answers is not in the document]
Comment (jpmenil) [reply srcpkgs/ta-lib/template]: ack

**Comment by jpmenil** [comment]
sorry for the disagreement :(

File: srcpkgs/python3-ta-lib/template [hunk 11]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'python3-ta-lib'
+pkgname=python3-ta-lib
+version=0.4.17
+revision=1
+archs=noarch
```
Comment (Chocimier) [inline srcpkgs/python3-ta-lib/template]: > Module contains compiled part, so isn't noarch.

File: srcpkgs/python3-ta-lib/template [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+wrksrc="TA-Lib-${version}"
+build_style=python3-module
+pycompile_module="talib"
```
Comment (Chocimier) [inline srcpkgs/python3-ta-lib/template]: There is no need to set this from some time ago.

File: srcpkgs/python3-ta-lib/template [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+hostmakedepends="python-setuptools python3-setuptools"
```
Comment (Chocimier) [inline srcpkgs/python3-ta-lib/template]: python-setuptools was only needed for python2 variant

File: srcpkgs/ta-lib-devel/template [hunk 14]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'ta-lib-devel'
+pkgname=ta-lib-devel
```
Comment (Chocimier) [inline srcpkgs/ta-lib-devel/template]: Should be split to ta-lib and ta-lib-devel, like eg. libarchive. Then -devel goes to makedepends and lib to depends.

**Comment by jpmenil** [comment]
closing this one, since theses libs are a bit outdated.
I can rework it if someone really need it.

**Pull request closed without merging**